File: 0766-toeplitz-matrix/0766-toeplitz-matrix.java
class Solution {
    public boolean isToeplitzMatrix(int[][] m) {
        for(int i=0;i<m.length-1;i++){
            for(int j=0;j<m[0].length-1;j++){
                if(m[i][j]!=m[i+1][j+1])return false;
            }
        }
        return true;
    }
}

File: 0766-toeplitz-matrix/0766-toeplitz-matrix.cpp
class Solution {
public:
    bool isToeplitzMatrix(vector<vector<int>>& m) {
        int i=0,j=m[0].size()-1;
        while(i<m.size()){
            int x,y;
            x=i,y=j;
            int cur=m[x][y];
            while(x<m.size()&&y<m[0].size()){
                if(m[x][y]!=cur)return false;
                x++,y++;
            }
            if(j==0){
                i++;
            }else{
                j--;
            }
        }
        return true;
    }
};